src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -649,8 +649,10 @@ export class MgtPeoplePicker extends MgtTemplatedComponent {
         if (this.defaultPeople) {
           people = this.defaultPeople;
         } else {
-          people = await getPeople(graph);
-          this.defaultPeople = people;
+          if (this.type === PersonType.Person || this.type === PersonType.Any) {
```

Comment:
Consider moving the check for type above line 648 and include a check whether group-id is set

Reply:
Shouldn't setting `defaultPeople` always take precedence over group type? That's why 
```
 if (this.defaultPeople) {
          people = this.defaultPeople;	       
        } else {	        
```
Then checking types.

Reply:
I came up with the following:
```
      if (!input.length && this._isFocused) {
        if (this.defaultPeople) {
          people = this.defaultPeople;
        } else {
          if (this.groupId) {
            if (this._groupPeople === null) {
              try {
                this._groupPeople = await getPeopleFromGroup(graph, this.groupId);
              } catch (_) {
                this._groupPeople = [];
              }
            }
            people = this._groupPeople || [];
          } else if (this.type === PersonType.Person || this.type === PersonType.Any) {
            people = await getPeople(graph);
          } else if (this.type === PersonType.Group) {
            const groups = (await findGroups(graph, '', this.showMax, GroupType.Any)) || [];
            people = groups;
          }
          this.defaultPeople = people;
        }
        this._showLoading = false;
      }
```

Reply:
Changes made here: 

2f7b3d6

Reply:
I think that should work